src/diskquota.c

```diff
@@ -525,7 +526,7 @@ disk_quota_worker_main(Datum main_arg)
 			if (!diskquota_is_paused())
 			{
 				/* Refresh quota model with init mode */
-				refresh_disk_quota_model(!MyWorkerInfo->dbEntry->inited);
+				refresh_disk_quota_model(MyWorkerInfo->dbEntry);
 				MyWorkerInfo->dbEntry->inited = true;
```

Comment:
I didn't quite understand why this assignment is not protected by a lock?

Reply:
Well, I guess the authors considered putting locks here to be an unnecessary action. Probably in order to avoid perfomance regress, because main work is done inside loop and asquiring locks is an expensive operation. Do you think we should add it here? There are lots of other places when we read and write these state variables without locks.